### PR TITLE
feat(sdk): update model

### PR DESCRIPTION
**Pull request by fsamin**
i will update worker model structure to be able to setup properly userdata, cmd (for freebsd per instance) and register os and architecture

is it fine for you ?

DO NOT MERGE THIS PR

### CODE REVIEW

File: sdk/worker.go [hunk 1]
```diff
@@ -83,34 +83,42 @@ type SpawnErrorForm struct {
...
+	ID                           int64               `json:"id" db:"id" cli:"-"`
+	Name                         string              `json:"name"  db:"name" cli:"name"`
+	Description                  string              `json:"description"  db:"description" cli:"description"`
+	Type                         string              `json:"type"  db:"type" cli:"type"`
+	ModelVirtualMachine          ModelVirtualMachine `json:"model_virtual_machine" db:"-" cli:"-"`
+	ModelDocker                  ModelDocker         `json:"model_docker" db:"-" cli:"-"`
+	Communication                string              `json:"communication"  db:"communication" cli:"communication"`
+	Disabled                     bool                `json:"disabled"  db:"disabled" cli:"disabled"`
+	Restricted                   bool                `json:"restricted"  db:"restricted" cli:"restricted"`
+	RegisteredBinaryCapabilities []string            `json:"registered_binary_capabilitties"  db:"-" cli:"-"`
```
Comment (yesnault) [inline sdk/worker.go]: registered_binary_capabilitties -> registered_binary_capabilities

File: sdk/worker.go [hunk 2]
```diff
@@ -83,34 +83,42 @@ type SpawnErrorForm struct {
...
 	Image    string `json:"os"`
 	Flavor   string `json:"flavor,omitempty"`
 	UserData string `json:"user_data"`
 }
+
+// ModelDocker for swarm, marathon and kubernetes
+type ModelDocker struct {
+	Image  string `json:"os"`
+	Memory int    `json:"memory"`
+	Cmd    string
```
Comment (yesnault) [inline sdk/worker.go]: missing json:"cmd", no?

Comment (yesnault) [reply sdk/worker.go]: otherwise, LGTM

File: sdk/worker.go [hunk 3]
```diff
@@ -83,34 +83,42 @@ type SpawnErrorForm struct {
...
+// ModelVirtualMachine for openstack or vsphere
+type ModelVirtualMachine struct {
 	Image    string `json:"os"`
 	Flavor   string `json:"flavor,omitempty"`
 	UserData string `json:"user_data"`
 }
+
+// ModelDocker for swarm, marathon and kubernetes
+type ModelDocker struct {
+	Image  string `json:"os"`
```
Comment (sguiheux) [inline sdk/worker.go]: maybe time to change json  os => image?

**Pull request closed without merging**